lib/core/shared/store.js

```diff
@@ -114,17 +135,101 @@ class Store {
+  createCollections (
+    collections,
+    { indexCacheOnly = false } = {}
+  ) {
+    return promiseAllN(
+      Object.entries(collections).map(
+        ([collection, config]) => async () => {
+          // @deprecated
+          if (config.mappings !== undefined && config.settings !== undefined) {
+
+            // @deprecated
+            const isConfigDeprecated = config.settings.number_of_shards === undefined && config.settings.number_of_replicas === undefined;
+
+            if (indexCacheOnly) {
+              return global.kuzzle.ask(
+                `core:storage:${this.scope}:collection:create`,
+                this.index,
+                collection,
+                // @deprecated
+                isConfigDeprecated ? { mappings: config.mappings } : config,
+                { indexCacheOnly });
+            }
+
+            const exist = await global.kuzzle.ask(
+              `core:storage:${this.scope}:collection:exist`,
+              this.index,
+              collection);
+
+            if (exist) {
+              // @deprecated
+              const dynamicSettings = (isConfigDeprecated)
+                ? null
+                : getESIndexDynamicSettings(config.settings);
+
+              const existingSettings = await global.kuzzle.ask(
+                `core:storage:${this.scope}:collection:settings:get`,
+                this.index,
+                collection);
+
+              if (! isConfigDeprecated
+                && parseInt(existingSettings.number_of_shards) !== config.settings.number_of_shards
+              ) {
+                if (global.NODE_ENV === 'development') {
+                  throw kerror.get(
+                    'storage',
+                    'wrong_collection_number_of_shards',
+                    collection,
+                    this.index,
+                    this.scope,
+                    'number_of_shards',
+                    config.settings.number_of_shards,
+                    existingSettings.number_of_shards);
+                }
+                global.kuzzle.log.warn([
+                  'Attempt to recreate',
+                  `an existing collection ${collection}`,
+                  `of index ${this.index} of scope ${this.scope}`,
+                  'with non matching',
+                  'static setting : number_of_shards',
+                  `at ${config.settings.number_of_shards} while existing one`,
+                  `is at ${existingSettings.number_of_shards}`,
+                ].join('\n'));
+              }
+
+              // Create call will update collection and cache it internally
+              return global.kuzzle.ask(
+                `core:storage:${this.scope}:collection:create`,
+                this.index,
+                collection,
+                // @deprecated
+                (isConfigDeprecated)
+                  ? { mappings: config.mappings }
+                  : { mappings: config.mappings, settings: dynamicSettings },
+              );
```

Comment:
I don't get the purpose of all of this. IMO if a collection already exists we only need to reject the collection creation process didn't we?